test/test_optimality_conditions.jl

```diff
@@ -21,6 +21,25 @@ check_optimality(f::IndBallL1, x, gamma, y) = begin
     return all(sign_is_correct) && check_optimality(IndSimplex(f.r), abs.(x), gamma, abs.(y))
 end
 
+check_optimality(f::NormTV, x, gamma, y) = begin
+    N = length(x)
+    # compute dual solution
+    u = zeros(N+1)
```

Comment:
You could enforce `u` this to have elements of the same type as `x` (or `y`). To do this you can do

```julia
R = real(eltype(x))  # or y
u = zeros(R, N+1)
```